### ISSUE TITLE
2.16 only: incorrect dependency check in the sample program ecdh_curve25519

### ISSUE THREAD

**Issue by gilles-peskine-arm** [Mbed-TLS/mbedtls#3242]
### Description
- Type: Bug
- Priority: Minor

---------------------------------------------------------------
## Bug

In Mbed TLS 2.16, in the default configuration:
```
programs/pkey/ecdh_curve25519
MBEDTLS_ECDH_C and/or MBEDTLS_ECDH_LEGACY_CONTEXT and/or MBEDTLS_ECP_DP_CURVE25519_ENABLED and/or MBEDTLS_ENTROPY_C and/or MBEDTLS_CTR_DRBG_C not defined
```

[Reported and analyzed by @josuah](https://github.com/ARMmbed/mbedtls/commit/43f564f29d7665e278783ae8ae32161be1ac02db#commitcomment-38728116).

Suggestion for non-regression: make a demo script that just runs this program so that it gets run on CI once https://github.com/ARMmbed/mbedtls/pull/2698 is merged and backported.

This bug was not present in 2.7 and was accidentally fixed in 2.18.

**Comment by danh-arm** [comment]
This appears to be a duplicate of https://github.com/ARMmbed/mbedtls/issues/3191. The reporter there says they will submit a fix.

**Comment by gilles-peskine-arm** [comment]
Mbed TLS 2.16 is no longer maintained.
